server/core/src/main/java/org/infinispan/server/core/admin/AdminServerTask.java

```diff
@@ -35,8 +35,18 @@ public String getType() {
    }
 
    public final T execute(TaskContext taskContext) {
-      Map<String, byte[]> rawParams = (Map<String, byte[]>) taskContext.getParameters().get();
-      Map<String, String> parameters = rawParams.entrySet().stream().collect(Collectors.toMap(Map.Entry::getKey, entry -> new String(entry.getValue(), CharsetUtil.UTF_8)));
+      Map<String, ?> raw = taskContext.getParameters().orElse(Collections.emptyMap());
+      Map<String, String> parameters = raw.entrySet().stream().collect(
```

Comment:
why not calling "map" method to transform

Reply:
I'm not going to do further refactorings to this code in this PR.